### PR TITLE
Sync the flink-provided profile in docs and testing suite comments

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Sync the flink-provided profile in docs and testing suite comments.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
           - java: 11
             spark: '3.1'
             spark-hadoop: '3.2'
-            profiles: '-DskipTests -Pspark-provided'
+            profiles: '-DskipTests -Pspark-provided -Pflink-provided'
```
Comment (pan3793) [inline .github/workflows/master.yml:57]: '-Pspark-provided -Pflink-provided' => '-Pflink-provided,spark-provided'

File: docs/develop_tools/distribution.md [hunk 2]
```diff
@@ -48,10 +49,10 @@ For instance,
 
 This results in a Kyuubi distribution named `kyuubi-{version}-bin-custom-name.tgz` for you.
 
-If you are planing to deploy Kyuubi where `spark` is provided, in other word, it's not required to bundle spark binary, use 
+If you are planing to deploy Kyuubi where `spark`/`flink` is provided, in other word, it's not required to bundle spark/flink binary, use 
 
 ```bash
-./build/dist --tgz --spark-provided
+./build/dist --tgz --spark-provided --flink-provided
```
Comment (turboFei) [inline docs/develop_tools/distribution.md:55]: keep the order as before

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -54,7 +54,7 @@ jobs:
           - java: 11
             spark: '3.1'
             spark-hadoop: '3.2'
-            profiles: '-DskipTests -Pspark-provided'
+            profiles: '-DskipTests -Pflink-provided -Pspark-provided'
```
Comment (pan3793) [inline .github/workflows/master.yml:57]: let's concat them by ,

File: .github/workflows/publish-snapshot.yml [hunk 4]
```diff
@@ -37,14 +37,14 @@ jobs:
         ASF_PASSWORD: ${{ secrets.NEXUS_PW }}
       run: |
         rm -rf ~/.m2/repository/org/apache/kyuubi
-        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pspark-provided
-        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pspark-provided,spark-3.1 -pl dev/kyuubi-extension-spark-3-1 -am
-        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pspark-provided,spark-3.2 -pl dev/kyuubi-extension-spark-3-2 -am
+        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pflink-provided -Pspark-provided
+        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pflink-provided -Pspark-provided,spark-3.1 -pl dev/kyuubi-extension-spark-3-1 -am
+        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pflink-provided -Pspark-provided,spark-3.2 -pl dev/kyuubi-extension-spark-3-2 -am
```
Comment (pan3793) [inline .github/workflows/publish-snapshot.yml:42]: ditto

File: .github/workflows/publish-snapshot.yml [hunk 5]
```diff
@@ -37,14 +37,14 @@ jobs:
...
     - name: Publish snapshot - branch-1.3
       if: ${{ matrix.branch == 'branch-1.3' }}
       env:
         ASF_USERNAME: ${{ secrets.NEXUS_USER }}
         ASF_PASSWORD: ${{ secrets.NEXUS_PW }}
       run: |
         rm -rf ~/.m2/repository/org/apache/kyuubi
-        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pspark-provided,kyuubi-extension-spark-3-1
+        ./build/mvn clean deploy -s ./build/release/asf-settings.xml -DskipTests -Pflink-provided -Pspark-provided,kyuubi-extension-spark-3-1
```
Comment (pan3793) [inline .github/workflows/publish-snapshot.yml:50]: ditto

**Review by pan3793 — APPROVED** [review]
LGTM, with minor comments

**Comment by ulysses-you** [comment]
shall we merge this PR after cut branch-1.4 ? since flink is tagged to 1.5.0

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
addressed the comments and will merge it after GA passed

**Comment by turboFei** [comment]
thanks for your review, merging to master

**Pull request closed without merging**